tests/cpp/SocketTests.cpp

```diff
@@ -50,47 +50,33 @@ namespace ut = boost::unit_test;
 
 BOOST_GLOBAL_FIXTURE( MinimalGlobalQtApp );
 
-BOOST_AUTO_TEST_CASE( testSocketConnectionValidWhenReturnedCorrectNetworkProtocolVersion )
+void testSocketConnect( const int32_t versionOffset )
 {
     QThread thread;
-    MockServer server;
-    server.moveToThread( &thread );
+    MockServer* server = new MockServer( NETWORK_PROTOCOL_VERSION + versionOffset );
```

Comment:
Any reason for making this a ptr? Looks equivalent to me, but slightly more complex with the added connection...